src/ansi-c/c_typecheck_argc_argv.cpp

```diff
@@ -42,6 +42,7 @@ void c_typecheck_baset::add_argc_argv(const symbolt &main_symbol)
     argc_symbol.type=op0.type();
     argc_symbol.is_static_lifetime=true;
     argc_symbol.is_lvalue=true;
+    argc_symbol.value = side_effect_expr_nondett(op0.type());
```

Comment:
This is surprising -- the convention elsewhere seems to be that a nil value implies nondet init (for example, Java globals)

Reply:
Is that maybe only the case in Java? The one place that takes care of initialising globals is `static_lifetime_init`, which will zero-initialise all symbols that don't have a value (unless they are marked `extern`). Now of course the Java front-end had to do this in its very own way...